File: Lab4/L4Q3.java
package Lab4;

import java.util.*;
public class L4Q3 {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int num , n = 0, sum = 0, ssum = 0, min = Integer.MAX_VALUE, max = Integer.MIN_VALUE;
		while(true) {
			System.out.print("Enter a score [negative score to quit]: ");
			if((num = sc.nextInt()) < 0) break;
			sum += num; ssum += num * num; n++;
			if(num < min) min = num;
			if(num > max) max = num;
		}
		sc.close();

		double avg = sum / n;
		double var = (ssum - Math.pow(sum, 2) / n) / (double) (n - 1);
		double std = Math.pow(var, 0.5);

		System.out.println("Minimum Score: " + min);
		System.out.println("Maximum Score: " + max);
		System.out.printf("Average Score: %.2f\n", avg);
		System.out.printf("Standard Deviation: %.2f\n ", std);
	}
}

File: umwix1002-TypeScript/lab/lab04/l4q3.ts
import {createConsoleInput} from "../utils/console-input";

async function l4q3Main(): Promise<void> {
    const consoleInput = createConsoleInput();

    const scores: number[] = [];

    while (true) {
        const score = await consoleInput.promptInteger("Enter a score [negative score to quit]: ");
        if (score < 0) {
            break;
        }
        scores.push(score);
    }

    if (scores.length === 0) {
        console.log("No scores entered.");
        return;
    }

    let sum: number = 0;
    let sumOfSquares: number = 0;
    let min: number = Infinity;
    let max: number = -Infinity;

    for (const score of scores) {
        sum += score;
        sumOfSquares += score * score;
        min = Math.min(min, score);
        max = Math.max(max, score);
    }

    const average: number = sum / scores.length;
    const variance: number = (sumOfSquares - sum * sum / scores.length) / (scores.length - 1);
    const standardDeviation: number = Math.sqrt(variance);

    console.log(`Sum: ${sum}`);
    console.log(`Min: ${min}`);
    console.log(`Max: ${max}`);
    console.log(`Average: ${average}`);
    console.log(`Standard Deviation: ${standardDeviation}`);
}

l4q3Main()
    .then(() => process.exit(0))
    .catch((error: unknown) => {
        console.error(error);
        process.exit(1);
    })